src/components/Environments/EnvironmentDetailsRoute.tsx

```diff
@@ -0,0 +1,194 @@
+import React, { lazy, Suspense, useCallback, useRef, useEffect, useState } from 'react'
+import { Switch, Route, Redirect, NavLink } from 'react-router-dom'
+import { ErrorBoundary, Progressing, BreadCrumb, useBreadcrumb, showError, useAsync } from '../common'
+import { useParams, useRouteMatch, useHistory, generatePath, useLocation } from 'react-router'
+import ReactGA from 'react-ga4'
+import { URLS } from '../../config'
+import PageHeader from '../common/header/PageHeader'
+import EnvConfig from './EnvironmentConfig/EnvConfig'
+import { getEnvAppList } from './EnvironmentListService'
+import EnvironmentOverview from './EnvironmentOverview/EnvironmentOverview'
+import { EnvSelector } from './EnvSelector'
+import ResourceListEmptyState from '../ResourceBrowser/ResourceList/ResourceListEmptyState'
+import EmptyFolder from '../../assets/img/Empty-folder.png'
+
+export default function EnvironmentDetailsRoute() {
+    const { path } = useRouteMatch()
+    const { envId } = useParams<{ envId }>()
+    const [envName, setEnvName] = useState('')
+    const [showEmpty, setShowEmpty] = useState<boolean>(true)
+    const [loading, envList] = useAsync(() => getEnvAppList({}), [])
+
+    useEffect(() => {
+        if (envList?.result) {
+            const environment = envList.result.envList?.find((env) => env.id === +envId)
+            setEnvName(environment.environment_name)
+            setShowEmpty(!environment.appCount)
+        }
+    }, [envList])
+
+    const renderRoute = () => {
+        if (loading)
+            return ( 
+                <div className='empty-state flex w-100'>
+                    {showEmpty ? <Progressing pageLoader /> : <ResourceListEmptyState
+                        imgSource={EmptyFolder}
+                        title="No applications available"
+                        subTitle="You don’t have access to any application in this app group."
+                    />}
+                </div>
+            )
+        return (
+            <ErrorBoundary>
+                <Suspense fallback={<Progressing pageLoader />}>
+                    <Switch>
+                        <Route path={`${path}/${URLS.APP_DETAILS}`}>
+                            <div>Env detail</div>
+                        </Route>
+                        <Route path={`${path}/${URLS.APP_OVERVIEW}`}>
+                            <EnvironmentOverview />
+                        </Route>
+                        <Route path={`${path}/${URLS.APP_TRIGGER}`}>
+                            <div>Build & Deploy</div>
```

Comment:
Are routes `DETAILS` & `TRIGGER` going to be replaced?

Reply:
will work in next round